File: nodejs/src/helpers/createMatrix.ts

export default async (numRows: number, numCols: number, fillZeros: boolean = false) : Promise<number[][]> => {

    const matriz :number[][] = [];
    for (let i = 0; i < numRows; i++) {
        const linha = [];
        for (let j = 0; j < numCols; j++) linha.push(fillZeros ? 0 : Math.floor(Math.random() * 10));
        matriz.push(linha);
    }
    return matriz;
}

File: golang/api/application/helpers/createMatrix.go
package helpers

import (
	"math/rand"
	"time"
)

func GenerateMatrix(numRows, numCols int, fillZeros bool) [][]int {
	matriz := make([][]int, numRows)
	source := rand.NewSource(time.Now().UnixNano())
	rng := rand.New(source)

	for i := 0; i < numRows; i++ {
		linha := make([]int, numCols)
		for j := 0; j < numCols; j++ {
			if fillZeros {
				linha[j] = 0
			} else {
				linha[j] = rng.Intn(10)
			}
		}
		matriz[i] = linha
	}
	return matriz
}
